### PR TITLE
Reordered queue-demo README items and fine-tuned parameters

### PR DESCRIPTION
**Pull request by njtran**
#### In main README.md 
- removed some extra lines
- added ecr_login step in installation script
- added CLOUD_PROVIDER=aws as `./hack/quick-install.sh` wasn't grabbing environment variables properly.

#### In queue-demo README.md
- added a possibility for reserved-capacity metrics producer to watch the magic happen 
- Changed order to configure IRSA before manifest is applied (and before that the namespace is created)

#### other
- fine-tuned the parameters of the demo, so that each node can fit only 4 of the pods, and that it scales to max 10 nodes and 40 pods

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,15 +38,15 @@ trap "rm -rf $TMP" EXIT
     cd karpenter
     make toolchain
     make generate
-    ./hack/quick-install.sh
+    aws ecr get-login-password --region ${REGION} | docker login --username AWS --password-stdin $KO_DOCKER_REPO
```
Comment (ellistarn) [inline README.md:41]: Ah -- we shouldn't be requiring this. We should just have the user use our v0.1.0 manifests. Do you want to copy the helm install scripts into here and then a kubectl apply -f the v0.1.0? That will remove any sort of ECR action, which will be a huge PITA for the user.

File: queue-length/README.md [hunk 2]
```diff
@@ -4,38 +4,63 @@ This demo will use the length of an AWS SQS Queue to autoscale a Deployment and
 
 We will deploy pods that handle 1 message every 30 seconds. The autoscalers are configured to create 4 nodes per pod.
 
-## Apply YAML and Watch
 
+## Create Demo Namespace
 ```bash
-wget https://raw.githubusercontent.com/ellistarn/karpenter-aws-demo/main/queue-length/manifest.yaml
+wget https://raw.githubusercontent.com/ellistarn/karpenter-aws-demo/main/queue-length/namespace.yaml
```
Comment (ellistarn) [inline queue-length/README.md:10]: namespace.yaml is a misleading name.

File: queue-length/README.md [hunk 3]
```diff
@@ -4,38 +4,63 @@ This demo will use the length of an AWS SQS Queue to autoscale a Deployment and
...
 watch 'kubectl get pods -l app=subscriber -n karpenter-queue-length-demo'
 watch 'kubectl get nodes -n karpenter-queue-length-demo'
 watch -d 'kubectl get metricsproducer demo -n karpenter-queue-length-demo -ojson | jq ".status.queue"'
 watch -d 'kubectl get horizontalautoscalers.autoscaling.karpenter.sh capacity -n karpenter-queue-length-demo -ojson | jq ".status" | jq "del(.conditions)"'
 watch -d 'kubectl get horizontalautoscalers.autoscaling.karpenter.sh subscriber -n karpenter-queue-length-demo -ojson | jq ".status" | jq "del(.conditions)"'
 watch -d 'kubectl get scalablenodegroup capacity -n karpenter-queue-length-demo -ojson | jq "del(.status.conditions)" | jq ".spec, .status"'
+
+wget https://raw.githubusercontent.com/ellistarn/karpenter-aws-demo/main/queue-length/manifest.yaml
+
+QUEUE_NAME=$USER-demo-queue \
```
Comment (ellistarn) [inline queue-length/README.md:41]: Why did you move this? Now the metrics producer is created before the queue is, which is a weird demo experience. I think it's fine where it is.

File: queue-length/README.md [hunk 4]
```diff
@@ -4,38 +4,63 @@ This demo will use the length of an AWS SQS Queue to autoscale a Deployment and
...
+QUEUE_ARN=arn:aws:sqs:$REGION:$AWS_ACCOUNT_ID:$QUEUE_NAME \
+QUEUE_URL=$(aws sqs create-queue --queue-name $QUEUE_NAME --output json | jq -r '.QueueUrl') \
+NODE_GROUP_ARN=$(aws eks describe-nodegroup --nodegroup-name demo --cluster-name ${CLUSTER_NAME} --output json | jq -r ".nodegroup.nodegroupArn") \
+envsubst < manifest.yaml | kubectl apply -f -
+
+```
+
+## Apply Additional YAML to watch node usage
```
Comment (ellistarn) [inline queue-length/README.md:49]: Just include this in the same YAML but put a comment on it, no need to make a special section on just this thing.

File: queue-length/README.md [hunk 5]
```diff
@@ -4,38 +4,63 @@ This demo will use the length of an AWS SQS Queue to autoscale a Deployment and
...
+kubectl apply -f reserved-capacity.yaml
+
+watch "kubectl get metricsproducers capacity-watcher -n karpenter-queue-length-demo -ojson | jq -r '.status.reservedCapacity'"
 ```
 
 ## Send messages to the Queue
 
 ```bash
 # Send 10 messages to the queue every 10 seconds
+QUEUE_NAME=$USER-demo-queue
```
Comment (ellistarn) [inline queue-length/README.md:62]: Let's leave this where is. QUEUE_NAME is also defined twice, now

File: queue-length/manifest.yaml [hunk 6]
```diff
@@ -57,7 +52,7 @@ spec:
           resources:
             requests:
               memory: 100Mi
-              cpu: 1
+              cpu: 1700m #1700m handles current configuration of 4 pods/node, lower this for higher pod density
```
Comment (ellistarn) [inline queue-length/manifest.yaml:55]: I'd specifically comment the machine type here.

File: README.md (hunk 1 above)
Comment (njtran) [reply README.md:41]: My understanding of the install script is that the helm install script is not the problem here. Doing `./hack/quick-install.sh` does `make apply` where the `ko` command requires us to be logged into a repo. If I'm misunderstanding this, and there's a way around doing ecr_login, I would love to hear it.

File: queue-length/README.md (hunk 2 above)
Comment (njtran) [reply queue-length/README.md:10]: `namespace.yaml` now only makes the namespace, so I think this fits here.

File: queue-length/README.md (hunk 3 above)
Comment (njtran) [reply queue-length/README.md:41]: I think the metrics producer is still made when the queue is made. The queue is made in now line 43. I moved around the commands because in the previous version, we were creating all the resources before we deployed the IRSA, which when I did the flow, resulted in some authentication issues. To make it clear that we do the IRSA step before adding any resources, I made this the order.

File: queue-length/README.md (hunk 4 above)
Comment (njtran) [reply queue-length/README.md:49]: Ok, sounds good. I thought it might be a bit confusing to have that extra metrics producer that isn't the queue-length or isn't being used for anyone who is just reading over the demo.

File: queue-length/README.md (hunk 5 above)
Comment (njtran) [reply queue-length/README.md:62]: I put it here because we had the `QUEUE_NAME=$blah \` up above. This makes it so that the variable is only defined for that function, and not created as an environment variable. I think I can just get rid of the \ above, and this will be fixed though.

File: queue-length/manifest.yaml (hunk 6 above)
Comment (njtran) [reply queue-length/manifest.yaml:55]: Got it! Good idea.

File: queue-length/README.md (hunk 4 above)
Comment (ellistarn) [reply queue-length/README.md:49]: Comments should make this clear.

File: queue-length/README.md (hunk 5 above)
Comment (ellistarn) [reply queue-length/README.md:62]: I don't think envsubst will work anymore if you do this.

File: queue-length/README.md [hunk 7]
```diff
@@ -50,7 +68,13 @@ sleep 10;
 done
 ```
 
-## Cleanup
+# Cleanup
+
 ```bash
+rm manifest.yaml
```
Comment (ellistarn) [inline queue-length/README.md:74]: Deleting the namespace will catch all of these resources.

File: reserved-capacity/README.md [hunk 8]
```diff
@@ -27,6 +27,11 @@ REPLICAS=30 envsubst < inflate.yaml | kubectl apply -f -
 ```
 
 ## Cleanup
+
 ```bash
+rm manifest.yaml
```
Comment (ellistarn) [inline reserved-capacity/README.md:32]: likewise

File: reserved-capacity/README.md [hunk 9]
```diff
@@ -27,6 +27,11 @@ REPLICAS=30 envsubst < inflate.yaml | kubectl apply -f -
...
+rm inflate.yaml
```
Comment (ellistarn) [inline reserved-capacity/README.md:33]: iiuc inflate should be in the namespace, so it should be cleaned up automatically, too

**Review by ellistarn — APPROVED** [review]
Nice fixes thank you!
